.github/workflows/docker-build-and-compose.yml

```diff
@@ -0,0 +1,49 @@
+name: Docker Build and Compose
+
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'client/**'
+      - 'server/**'
+      - 'docker-compose.yml'
```

Comment:
Let's remove the [path qualifiers; they're overly verbose](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-including-and-excluding-paths). Why do we only care about changes to the `client/` and `server/` dirs and the `docker-compose.yml` file? We were already only concerned about running these if a `push` or `pull_request` is made to the `main` branch; that's sufficient.